crates/ide/src/completion/presentation.rs

```diff
@@ -191,6 +191,22 @@ impl Completions {
         func: hir::Function,
         local_name: Option<String>,
     ) {
+        fn add_arg(arg: &str, ty: &Type, ctx: &CompletionContext) -> String {
+            let mut prefix = "";
+            if let Some(derefed_ty) = ty.remove_ref() {
+                ctx.scope.process_all_names(&mut |name, scope| {
+                    if prefix != "" {
+                        return;
+                    }
+                    if let ScopeDef::Local(local) = scope {
+                        if name.to_string() == arg && local.can_unify(derefed_ty.clone(), ctx.db) {
```

Comment:
@flodiebold what do you think would be the best API, for an IDE, to check if one type is a reference + another type? I somewhat worry about exposing something called `can_unify`.

Reply:
Hmmm. I do think we should have a way of checking "can we put a value of this Type into something expecting this other Type". Not sure whether it should be called `can_unify`, or even directly use unification. With these fully-resolved types, there won't be any type variables left anyway, so mostly this is just going to be equality anyway, except for situations with unknown types, I think.

I don't think we should expose `can_unify` as a query though.

Reply:
Ah I see - didn't realize that they'd be already fully resolved at this point. Took out all the logic for `can_unify` and switched to just using equality on the types.